logger/options.go

```diff
@@ -51,27 +52,24 @@ func validateOptionLength(option string, value string, problems *[]string) {
 
 func (options *Options) validate() error {
 	var problems []string
-	reMacAddress := regexp.MustCompile(`^([0-9a-fA-F][0-9a-fA-F]:){5}([0-9a-fA-F][0-9a-fA-F])`)
 	reHostname := regexp.MustCompile(`(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]*[A-Za-z0-9])`)
-	reIPAddress := regexp.MustCompile(`(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}`)
 
 	validateOptionLength("App", options.App, &problems)
 	validateOptionLength("Env", options.Env, &problems)
 	validateOptionLength("Hostname", options.Hostname, &problems)
 	validateOptionLength("Level", options.Level, &problems)
 
-	if options.MacAddress != "" && (!reMacAddress.MatchString(options.MacAddress)) {
-		problems = append(problems, InvalidOptionMessage{"MacAddress", "invalid format"}.String())
+	if options.MacAddress != "" {
+		if _, err := net.ParseMAC(options.MacAddress); err != nil {
+			problems = append(problems, InvalidOptionMessage{"MacAddress", "invalid format"}.String())
```

Comment:
Appending all the errors into one string makes it difficult for the user to handle errors programatically. I think instead you can define a new error type that has an array of the bad fields. This will allow the users code to inspect each bad field and respond accordingly.

The `func Error() string` function can still print the same error message as before.

Reply:
we'll need to use the output of `ParseMac` to convert back to the "standard" representation, in case we are provided a different (but valid) format than we expect.

https://golang.org/src/net/mac.go

the `String()` method on HardwareAddr will ensure we get the right format.